core/os/android/adb/commands.go

```diff
@@ -132,6 +177,8 @@ func (b *binding) InstallAPK(ctx context.Context, path string, reinstall bool, g
 		args = append(args, "--force-queryable")
 	}
 	args = append(args, path)
+
+	b.Unroot(ctx)
```

Comment:
Comment why we do this, e.g.:

```
// Installing an APK as root fails on devices with a corp account,
// so make sure to unroot before installing here. In theory there's
// a potential race with a concurrent trace or package listing that
// would call b.Root() concurrently, but in practice this is very
// unlikely to happen.
```

Reply:
I thought about the race condition but have no nice solution for it. The root/unroot state exists outside the program and can be modified by any program or even the user at any time. Someone could adb unroot in a terminal at a bad time and cause the same problem you're describing above.

Reply:
Agreed in the general case.

We cannot ensure an exclusive access over adb, but AGI code assumes that no other program/user is using adb at the same time AGI runs. Our troubleshooting documentation (https://gpuinspector.dev/docs/troubleshooting) says:

```
- Stop any program that may interact with the device over ADB, such as Android Studio.
```

I think just putting a comment there is OK for this use-case.